File: acoustics_toolbox_r2020/Python/Bellhop/Tests/DW/ParaBot/ParaBot.m
clear all, close all

addpath /home/orodrig/FORdoc/at/Matlab/Plot
addpath /home/orodrig/FORdoc/at/Matlab/ReadWrite

disp('Bellhop - Parabolic boundaries:') 
disp('ray calculations:') 

fid = fopen('ParaBot.ati','r');
s = fscanf(fid,'%s\n',[1 1]);
n = fscanf(fid,'%d\n',[1 1]);
data = fscanf(fid,'%f %f\n',[2 n]);
fclose(fid);

ratikm = data(1,:); rati = ratikm*1000; 
zati   = data(2,:);

fid = fopen('ParaBot.bty','r');
s = fscanf(fid,'%s\n',[1 1]);
n = fscanf(fid,'%d\n',[1 1]);
data = fscanf(fid,'%f %f\n',[2 n]);
fclose(fid);

rbtykm = data(1,:); rbty = rbtykm*1000; 
zbty   = data(2,:);

system('bellhop.exe ParaBot');

plotray('ParaBot.ray')
hold on
plot(rati,zati,'b','LineWidth',2)
plot(rbty,zbty,'k','LineWidth',2)
hold off

disp('done.')

File: acoustics_toolbox_r2020/Python/Bellhop/Tests/DW/ParaBot/ParaBot.py
#=======================================================================
# 
# Bellhop: Parabolic bottom profile
# Mexilhoeira Grande, Dom Jul 30 19:37:05 WEST 2017
# Written by Tordar 
# 
#=======================================================================

from os import system
from numpy import *
from scipy.io import *
from pylab import *
sys.path.append ("/home/orodrig/FORdoc/at/Python/")
from plotray import *

print("Bellhop - Parabolic bottom profile:") 
print("Ray trace run")
print("Geometric hat beams in Cartesian coordinates")

fid = open('ParaBot.ati','r')
theline = fid.readline()
theline = fid.readline()
n       = int( theline )
ratikm  = zeros( n ) 
zati    = zeros( n )
for i in range(n):
    theline = str( fid.readline() )
    datai = theline.split()
    ratikm[ i] = float( datai[0] )
    zati[   i] = float( datai[1] )
fid.close()

rati = ratikm*1000

fid = open('ParaBot.bty','r')
theline = fid.readline()
theline = fid.readline()
n       = int( theline )
rbtykm  = zeros( n ) 
zbty    = zeros( n )
for i in range(n):
    theline = str( fid.readline() )
    datai = theline.split()
    rbtykm[ i] = float( datai[0] )
    zbty[   i] = float( datai[1] )
fid.close()

rbty = rbtykm*1000

system("bellhop.exe ParaBot")

figure(1)
plotray('ParaBot.ray')
plot(rati,-zati,'b',linewidth=2)
plot(rbty,-zbty,'k',linewidth=2)
xlabel('Range (m)')
ylabel('Depth (m)')
title("Bellhop - Parabolic bottom profile")

show()

print("done.")

